INN/Largo
# Remove the Staff List Widget, [roster] shortcode, and related user meta forms

## Changes

This pull request makes the following changes:

- Removes the Largo Staff List widget
- Removes the `[roster]` shortcode
- Deprecates two functions used for same

## To Do

- [x] ~Because the meta boxes used by the Staff List Widget and the `[roster]` shortcode are also used for bylines, specifically the "job title" meta and the "show_email" meta, we're going to have to figure out how Largo and this plugin interact. Who is responsible for providing the form?~
    - ~I think this might work best if Largo keeps the complete form, but unregisters the plugin's corresponding form functions. The metadata would remain the same across both the theme and the plugin.~
    - obsoleted by the fact that the plugin will not exist.
- [x] Clean out tests for deprecated functions
- [x] Deprecate functions that are no longer used in Largo


## Why

Because we're pulling this functionality out of Largo as part of https://github.com/INN/largo/issues/1505 and https://github.com/INN/staff

## Testing/Questions

Features that this PR affects:

- https://github.com/INN/staff
- The Largo Staff Roster widget
- the `[roster]` shortcode


Questions that need to be answered before merging:

- ~[ ] do we need to make any updates to https://github.com/INN/staff ?~ Per [the 2020-03-03 planning proposals doc](https://docs.google.com/document/d/1G1Tgf5P-9GMoMr1EelB4arsq9KsdX4dEdN6psXNLgf8/edit?ts=5e619d67#) we are cancelling this plugin
- ~[ ] does the migration function work?~
- [ ] do we want to recommend plugin alternatives to this deprecated functionality?

Steps to test this PR:

1. On a site with many users, and running `trunk`:
2.  add the `[roster]` shortcode to a page
2. Add the Largo Staff Roster widget to a widget area.
3. Check out this branch.
4. View the page that has the `[roster]` shortcode: the shortcode should be visible
5. View the widget area containing the Largo Staff Roster widget: it should be gone
